tests/unit/packaging.t

```diff
@@ -655,6 +655,56 @@ is_deeply(
 	]
 ) or diag explain $product_ref->{packagings};
 
-#
+# Agregate components by parent materials
+
+$product_ref = {
+	lc => "en",
+	product_quantity => 200,
+	packagings => [
+		{
+			material => 'pet',
+		},
+		{
+			material => 'paper',
+			weight_specified => 10,
+		},
+		{
+			material => 'steel',
+			weight_measured => 100,
+			number_of_units => 5,
+		},
+		{
+			material => 'aluminium',
+			weight_measured => 10,
+			number_of_units => 4,
+		},
+	],
+};
+
+ProductOpener::Packaging::canonicalize_packaging_components_properties($product_ref);
+ProductOpener::Packaging::aggregate_packaging_by_parent_materials($product_ref);
+
+is_deeply(
+	$product_ref->{packagings_materials},
+	{
+		'all' => {
+			'weight' => 550,
+			'weight_100g' => '275',
+			'weight_percent' => 100
+		},
+		'en:metal' => {
+			'weight' => 540,
+			'weight_100g' => '270',
+			'weight_percent' => '98.1818181818182'
+		},
+		'en:paper-or-cardboard' => {
+			'weight' => 10,
+			'weight_100g' => '5',
+			'weight_percent' => '1.81818181818182'
+		},
+		'en:plastic' => {}
+	}
+
+) or diag explain $product_ref->{packagings_materials};
```

Comment:
Would be good to add at least one test with an unknown material.

I always like to also test with an empty hash.